encodings/fastlanes/src/bitpacking/compute/search_sorted.rs

```diff
@@ -0,0 +1,114 @@
+use std::cmp::Ordering;
+use std::cmp::Ordering::Greater;
+
+use fastlanes::BitPacking;
+use vortex::array::primitive::PrimitiveArray;
+use vortex::array::sparse::SparseArray;
+use vortex::compute::search_sorted::{
+    search_sorted, IndexOrd, Len, SearchResult, SearchSorted, SearchSortedFn, SearchSortedSide,
+};
+use vortex::{ArrayDType, ArrayTrait, IntoArrayVariant};
+use vortex_dtype::{match_each_unsigned_integer_ptype, NativePType};
+use vortex_error::VortexResult;
+use vortex_scalar::Scalar;
+
+use crate::{unpack_single_primitive, BitPackedArray};
+
+impl SearchSortedFn for BitPackedArray {
+    fn search_sorted(&self, value: &Scalar, side: SearchSortedSide) -> VortexResult<SearchResult> {
+        let ptype = self.ptype();
+        match_each_unsigned_integer_ptype!(ptype, |$P| {
+            let unwrapped_value: $P = value.cast(self.dtype())?.try_into().unwrap();
+            if let Some(patches_array) = self.patches() {
+                if (unwrapped_value.leading_zeros() as usize) < ptype.bit_width() - self.bit_width() {
```

Comment:
nit: BitPackedArray can probably use a `max_packed_value() -> usize` method

Reply:
Good idea, it's easier to reason about it in terms of max value and not leading_zeros